skfda/representation/basis.py

```diff
@@ -325,6 +325,70 @@ def _list_to_R(self, knots):
     def _to_R(self):
         raise NotImplementedError
 
+    def inner_matrix(self, other=None):
+
+        r"""Return the Inner Product Matrix of a pair of basis.
+
+        The Inner Product Matrix is defined as
+
+        .. math::
+            IP_{ij} = \langle\phi_i, \theta_j\rangle
+
+        where :math:`\phi_i` is the ith element of the basi and
+        :math:`\theta_j` is the jth element of the second basis.
+        This matrix helps on the calculation of the inner product
+        between objects on two basis and for the change of basis.
+
+        Args:
+            other (:class:`Basis`): Basis to compute the inner product
+            matrix. If not basis is given, it computes the matrix with
+            itself returning the Gramian Matrix
+
+        Returns:
+            numpy.array: Inner Product Matrix of two basis
+
+        """
+        if other is None or self == other:
+            return self.gramian_matrix()
+
+        first = self.to_basis()
+        second = other.to_basis()
+
+        gramian = np.zeros((self.nbasis, other.nbasis))
+
+        for i in range(self.nbasis):
+            for j in range(other.nbasis):
+                gramian[i, j] = first[i].inner_product(second[j], None, None)
+
+        return gramian
+
+    def gramian_matrix(self):
```

Comment:
I prefer `gram_matrix`. It is personal taste, however.

Reply:
Nice